docs/Design/SoftDetailedDes/MIS.tex

```diff
@@ -229,6 +228,92 @@ \subsubsection{Local Functions}
 
 \newpage
 
+\section{MIS of Location Management Module} \label{mLocation} 
+The Location Management Module manages user location data and outputs relevant resources based the available user location data.
+
+\wss{You can reference SRS labels, such as R\ref{R_Inputs}.}
+
+\wss{It is also possible to use \LaTeX for hypperlinks to external documents.}
+
+\subsection{Module}
+
+locationManagement
+
+\subsection{Uses}
+
+ Database Driver Module (Section \ref{mDBDriver})
+
+\subsection{Syntax}
+
+\subsubsection{Exported Constants}
+None
+
+\subsubsection{Exported Access Programs}
+
+\begin{center}
+\begin{tabular}{p{5cm} p{3cm} p{3cm} p{5cm}}
+\hline
+\textbf{Name} & \textbf{In} & \textbf{Out} & \textbf{Exceptions} \\
+\hline
+getLastLocation & - & String & noLocationSavedException \\
+setLastLocation & String & - & invalidLocationException \\
+getNearbyStores & - & List$<$String$>$ & - \\
+setRadius & $\mathbb{N}$ & - & - \\
+\hline
+\end{tabular}
+\end{center}
+
+\subsection{Semantics}
+
+\subsubsection{State Variables}
+None
+
+\subsubsection{Environment Variables}
+None
+
+\subsubsection{Assumptions}
+None
+
+\subsubsection{Access Routine Semantics}
+
+\noindent getLastLocation():
+\begin{itemize}
+\item transition: none
+\item output: \( \textit{out} \coloneqq location \), where textit{location} is a String representing the user location stored in the database (Section \ref{mDBDriver})
+\item exception: \( \textit{exc} \coloneqq noLocationSavedException \), if the database (Section \ref{mDBDriver}) has no value saved for user location
```

Comment:
you can probably express this part formally like `$getLastLocation() = null \Rightarrow noLocationSavedException$`. You could probably use None instead of null too. Potentially something similar could be done on the output by adding `$\neq getLastLocation() = null \Rightarrow$ return location, where textit{location} is a string...`